source/slang/slang-type-layout.cpp

```diff
@@ -479,8 +491,8 @@ struct CUDALayoutRulesImpl : DefaultLayoutRulesImpl
         SimpleLayoutInfo vectorInfo;
         vectorInfo.kind = elementInfo.kind;
         vectorInfo.size = elementInfo.size * elementCount;
-        vectorInfo.alignment = elementInfo.alignment;
-    
+        vectorInfo.alignment = getVectorAlignment(
```

Comment:
It looks like size/alignment on CUDA is unfortunately more subtle than this. Can be improved on in subsequent PR.